src/pool/srv_target.c

```diff
@@ -1528,14 +1528,24 @@ ds_pool_tgt_query_map_handler(crt_rpc_t *rpc)
 	D_DEBUG(DB_TRACE, DF_UUID": handling rpc %p\n",
 		DP_UUID(in->tmi_op.pi_uuid), rpc);
 
-	hdl = ds_pool_hdl_lookup(in->tmi_op.pi_hdl);
-	if (hdl == NULL) {
-		rc = -DER_NO_HDL;
-		goto out;
+	if (daos_rpc_from_client(rpc) &&
+	    !is_pool_from_srv(in->tmi_op.pi_uuid, in->tmi_op.pi_hdl)) {
+		hdl = ds_pool_hdl_lookup(in->tmi_op.pi_hdl);
+		if (hdl == NULL) {
+			rc = -DER_NO_HDL;
+			goto out;
+		}
+		pool = hdl->sph_pool;
+	} else if (!daos_rpc_from_client(rpc) &&
```

Comment:
I am confused between looking at this patch, another similar patch https://github.com/daos-stack/daos/pull/7082
and the associated tickets https://daosio.atlassian.net/browse/DAOS-9785 and https://daosio.atlassian.net/browse/DAOS-8870

Where can the engine code be found that sends a POOL_TGT_QUERY_MAP RPC? I only see that in pool map refresh code in daos/src/pool/cli.c that I assume is not called by engine code.

Reply:
> Where can the engine code be found that sends a POOL_TGT_QUERY_MAP RPC? I only see that in pool map refresh code in daos/src/pool/cli.c that I assume is not called by engine code.

EC aggregation (on engine) needs to fetch the data on another engine by using object client stack, which will use map_refresh. see agg_fetch_odata_cells()